src/server/game/Spells/Spell.cpp

```diff
@@ -2448,7 +2448,7 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
         // Needs to be called after dealing damage/healing to not remove breaking on damage auras
         DoTriggersOnSpellHit(spellHitTarget, mask);
 
-        // if target is fallged for pvp also flag caster if a player
+        // if target is flagged for pvp also flag caster if a player
         if (unit->IsPvP() && m_caster->GetTypeId() == TYPEID_PLAYER)
             m_caster->ToPlayer()->UpdatePvP(true);
```

Comment:
This is still throwing me for a loop. I've taken another look at it this morning. Some of my observations:

In `DoAllEffectOnTarget`, `spellHitTarget` must not be NULL in order to get at the only call to `DoSpellHitOnUnit`, where my fix is. There are many calls to `HandleEffects`, but only one call to `DoSpellHitOnUnit` in the entire project.

As far as I can see, the only way that `spellHitTarget` will be reset to NULL is if the return value is **NOT** `SPELL_MISS_NONE` (~Line 2299 of Spell.cpp). But if we've gotten to the point of calling `HandleEffects` to actually make the wheels spin on the spell, then the only thing that `DoSpellHitOnUnit` can return is `SPELL_MISS_NONE` (~Line 2653 of Spell.cpp).